base/structure.py

```diff
@@ -1112,63 +1170,188 @@ def by_offset(self, offset):
+
+        # Now we call our members_t.__members_at__ helper-method so that we can check the
+        # members that are returned to verify that they're within our search boundaries.
+        items, unionQ = [], owner.ptr.is_union
+        for mptr in self.__members_at__(offset):
+            mleft, mright = 0 if unionQ() else mptr.soff, mptr.eoff
+
+            # Check the offset is within our current member's boundaries, and add it to
+            # our list if it is so that we can count our results later.
+            if mleft <= offset < mright:
+                items.append(mptr)
+            continue
 
-        # Get that member's size, so that way we can really confirm that this
-        # offset isn't pointing to anything.
-        msize = idaapi.get_member_size(mptr)
-        if (offset < minimum) or (offset >= maximum + msize):
+        # If we didn't find any items, then we need to throw up an exception because
+        # we're unable to proceed any farther without any members to search through.
+        if not items:
             cls = self.__class__
-            raise E.OutOfBoundsError(u"{:s}({:#x}).members.by_realoffset({:+#x}) : Requested offset not within bounds {:#x}<>{:#x}.".format('.'.join([__name__, cls.__name__]), owner.id, offset, minimum, maximum + msize))
+            raise E.MemberNotFoundError(u"{:s}({:#x}).members.by_realoffset({:+#x}) : Unable to find member at the specified offset ({:+#x}).".format('.'.join([__name__, cls.__name__]), owner.id, offset, offset))
 
-        # Now we can get a pointer to an actual member
-        mem = idaapi.get_member(owner.ptr, offset)
-        if mem is None:
+        # If we found more than one result, then we need to warn the user about it
+        # because we're going to have to make a decision on their behalf. This really
+        # should only be happening when we're a union type.
+        if len(items) > 1:
             cls = self.__class__
-            raise E.MemberNotFoundError(u"{:s}({:#x}).members.by_realoffset({:+#x}) : Unable to find member at specified offset ({:+#x}).".format('.'.join([__name__, cls.__name__]), owner.id, offset, offset))
+            iterable = (idaapi.get_member_fullname(mptr.id) for mptr in items)
+            logging.warning(u"{:s}({:#x}).members.by_realoffset({:+#x}) : The requested offset ({:#x}) contains more than one member ({:s}).".format('.'.join([__name__, cls.__name__]), owner.id, offset, offset, ', '.join(map(utils.string.to, iterable))))
+
+            # Grab the type information for each member so we can determine if the
+            # requested offset points at an array or a structure.
+            candidates = []
+            for mptr in items:
+                opinfo = idaapi.opinfo_t()
+                res = idaapi.retrieve_member_info(mptr, opinfo) if idaapi.__version__ < 7.0 else idaapi.retrieve_member_info(opinfo, mptr)
```

Comment:
As the result of res is being explicitly used, add a comment describing how it's being used.